.github/release_pull_request_template.md

```diff
@@ -0,0 +1,42 @@
+This is the release pull request for {version} of the WooCommerce Blocks plugin.
+
+## Communication
+
+<!--
+  This section is for any notes related to communicating the release.
+  Please include any extra details with each item as needed.
+-->
+
+* [ ] What is being introduced in this release?
+* [ ]  If this release has potentially **high customer impact**, is a blog post (section) ready? Provide a link if applicable.
+* [ ] Is the **relevant developer documentation updated**? Please provide links if applicable.
+* [ ] Any special instructions or helpful notes for Happiness engineers are posted (this might not be needed for every release).
+
+## Quality
+
+<!--
+  This section is for any notes related to quality around the release.
+  Please include any extra details with each item as needed.
+-->
+
+* [ ] Have you **written tests** to cover the changes?
+     * [ ] Unit tests
+     * [ ] E2E tests
+     * [ ] for each supported WordPress and WooCommerce core versions.
+* [ ] Have you tested on the applicable platforms?
+     * [ ] mobile
+     * [ ] desktop
+* [ ] Does this release affect API's and conform to API versioning policy?
+* [ ] Have you considered the impact of the changes in the release on **other extensions** and **backward compatibility**?
```

Comment:
These should be part of the release notes.

Reply:
Right, it might be good for me to add a section on followup tasks (which include release notes etc). I think some tasks might need to be _written before_ the release, but _published after_ the release. So I might try to nail that down a bit clearer in this via comments.